### ISSUE TITLE
Version v0.3

### ISSUE THREAD

**Issue by olegklimov** [smallcloudai/refact-vs-classic#34]
#8 
#33 
#32 
#31 
#30 
#29 
#28 
#24 
#20

**Comment by oliverqx** [comment]
@olegklimov can I get assigned?

**Comment by gechdejenb** [comment]
can I get assigned for the issue?

**Comment by digital-phoenix** [comment]
@olegklimov I'd be fine with @gechdejenb taking over this issue. I have enough other stuff to work on.

**Comment by Muhammad-Musab** [comment]
can i get assigned for this issue?

**Comment by olegklimov** [comment]
oh yeah sure why not

**Comment by korayaltq** [comment]
@Muhammad-Musab Are you still working for the issue? If not, I would love to work.

**Comment by RafaelJohn9** [comment]
Hello, here, requesting to be assigned this issue

**Comment by Ahmadkhan02** [comment]
I was trying to set this up, but how am I supposed to run the workflow for 'refact-lsp.exe' (it asks for admin rights)

**Comment by olegklimov** [comment]
hi @Ahmadkhan02 
It doesn't require any special rights, works on my Windows box just fine without asking anything. Maybe it asks to open a TCP port, but then it's 127.0.0.1 so it's not even a port visible from outside 🤷

**Comment by amiransari001** [comment]
Hello, 

Not sure if Muhammad-Musab is still working on this issue. If so, you can ignore this. If not, I would love to be assigned. 

I don’t have any experience with open source contributions or with VS Code extension development, but I have quite a bit of experience in Python, JavaScript, and working with open source projects. I also learn quickly and have a broad, deep, and solid foundation of knowledge, not to mention a bunch of time  on my hands. 

Appreciate your time.

**Comment by Ahmadkhan02** [comment]
> hi @Ahmadkhan02 It doesn't require any special rights, works on my Windows box just fine without asking anything. Maybe it asks to open a TCP port, but then it's 127.0.0.1 so it's not even a port visible from outside 🤷

had to delete refact-lsp.exe once, somehow ended up with a corrupt file earlier

**Comment by Ahmadkhan02** [comment]
/attempt https://github.com/smallcloudai/refact-vs-classic/issues/34

**Comment by Ahmadkhan02** [comment]
no, I am still facing issues with refact-lsp.exe. After git clone. it shows up like 
![Screenshot 2024-05-13 022156](https://github.com/smallcloudai/refact-vs-classic/assets/75062775/e8fa6e78-612c-415a-b519-e83e15ff67c0)
![Screenshot 2024-05-13 022236](https://github.com/smallcloudai/refact-vs-classic/assets/75062775/49aaf410-5ec1-4a79-a16f-b5ed8c016ab6)

**Comment by sushil026** [comment]
Attempt #34

**Comment by Ahmadkhan02** [comment]
> no, I am still facing issues with refact-lsp.exe. After git clone. it shows up like
> 
> ![Screenshot 2024-05-13 022156](https://private-user-images.githubusercontent.com/75062775/329879659-e8fa6e78-612c-415a-b519-e83e15ff67c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0OTUyNTAsIm5iZiI6MTcxNjQ5NDk1MCwicGF0aCI6Ii83NTA2Mjc3NS8zMjk4Nzk2NTktZThmYTZlNzgtNjEyYy00MTVhLWI1MTktZTgzZTE1ZmY2N2MwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDIwMDkxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiMjYxMTM4OTNiYWQ1ODI4NDk5MWMwNmVkYWFhYmZkMTNmMjcyODI0NjQxOGMxMWE0ZGI5ZjE5OWQ4MGZkMWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jxdxFVv9MyFBk11qvGoQFjYFr3ukFcl4ixeuEs3IGkE)
> 
> Also when running workflow from github cli:
> 
> ![Screenshot 2024-05-13 022236](https://private-user-images.githubusercontent.com/75062775/329879661-49aaf410-5ec1-4a79-a16f-b5ed8c016ab6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY0OTUyNTAsIm5iZiI6MTcxNjQ5NDk1MCwicGF0aCI6Ii83NTA2Mjc3NS8zMjk4Nzk2NjEtNDlhYWY0MTAtNWVjMS00YTc5LWExNmYtYjVlZDhjMDE2YWI2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTIzVDIwMDkxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxNDExNmE3ZDZhMDNjMjQyOTMxOTg3NmViOWI2OGE1MjI3MTY1MWYyMTRmNjM5MjhhMjNjMDQzNmEyNDZmN2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kXIda5gfpsJsNieC6XR-aFKk5O3iM3Z-7q5eLzVh7o0)

@olegklimov @digital-phoenix  any help on this? Would like to work on this issue but this is a major blocker

**Comment by d80ep08th** [comment]
@olegklimov , bro @Muhammad-Musab aint produced any work, @digital-phoenix removed the assignment in March. Seems like the issue is up for grabs, no ?

**Comment by Ahmadkhan02** [comment]
> hi @Ahmadkhan02 It doesn't require any special rights, works on my Windows box just fine without asking anything. Maybe it asks to open a TCP port, but then it's 127.0.0.1 so it's not even a port visible from outside 🤷

Got it working. Had to build the refact-lsp code separately. Working on the issues now

**Comment by mariomerida15** [comment]
@olegklimov can i get assigned?

And what i need to do

**Comment by Ahmadkhan02** [comment]
I have made some progress on the issues but cant get the app to work. I have tracked to issue down to AddFile in RefactLanguageClient.cs. Adding some debug: 
`public async Task AddFile(String filePath, String text)
{

    Debug.WriteLine( $"Adding file {filePath} to LSP");

    //wait for the rpc 
    while (this.Rpc == null) await Task.Delay(1);

    //dont send the file to the lsp if the lsp already knows about it
    if (ContainsFile(filePath))
    {
        return;
    }

    //message to send to lsp
    var openParam = new DidOpenTextDocumentParams
    {
        TextDocument = new TextDocumentItem
        {
            Uri = new Uri(filePath),
            LanguageId = filePath.Substring(filePath.LastIndexOf(".") + 1),
            Version = 0,
            Text = text
        }
    };

    //send message to lsp catch any communication errors
    try
    {;
        await Rpc.NotifyWithParameterObjectAsync("textDocument/didOpen", openParam);
        //add file to known file set
        files.Add(filePath);

        Debug.WriteLine($"File '{filePath}' added successfully");

    }
    catch (Exception e)
    {
        Debug.WriteLine("Exception: " + e.Message);
        ShowStatusBarError("Server Exception: \n" + e.Message);
    }
}`

i get error as: 
Adding file file:///C:/Users/kahma/AppData/Local/Temp/0u5gqz5n..js to LSP
19:01:57:732	Exception thrown: 'System.ObjectDisposedException' in mscorlib.dll
19:01:57:732	Exception: Cannot access a disposed object.

how do you guys get it running? 

NOTE: I have the refact-lsp.exe now. Had to build it seprate.

@olegklimov @digital-phoenix

**Comment by Ahmadkhan02** [comment]
@olegklimov anything for the above?

**Comment by olegklimov** [comment]
This issue is still valid!

@Ahmadkhan02 are you working on this?

**Comment by Ahmadkhan02** [comment]
Yup @olegklimov, still working on this. Actually I am mostly done with the changes but as I stated above, I just can’t get the extension to run and hence can’t test

Even a new clone doesn’t run

**Comment by rogueloop** [comment]
hey, @Ahmadkhan02 let's collab if you are ready.

**Comment by reymondzzzz** [comment]
@Ahmadkhan02 I'm here to help you. What the problem with extension do u have? we have backend binary with lsp https://github.com/smallcloudai/refact-lsp. Build this one and put it in resources. Where can I find your code and test it by myself?

**Comment by Ahmadkhan02** [comment]
Hi @reymondzzzz 

![Screenshot 2024-07-12 215508](https://github.com/user-attachments/assets/1392cd40-75fd-40bd-8834-701e7e786bf6)

I have already built and added refact-lsp as shown. Thats not the issue. I am starting to think my OS is the issue here. What OS do you guys use? I can see visual studio 2022 for mac is also a thing. I can get a mac to try if thats what you guys use

**Comment by reymondzzzz** [comment]
We use vsclassic on windows only. Microsoft doesn't no longer support mac version of VS

**Comment by Ahmadkhan02** [comment]
picking up this issue again after some time. Any idea when this issue comes up? 

 
![Screenshot 2024-08-26 021348](https://github.com/user-attachments/assets/c28e4675-43da-4f38-bfab-997f0706a357)

**Comment by bernardev254** [comment]
/attempt https://github.com/smallcloudai/refact-vs-classic/issues/34

**Comment by itsxdamdam** [comment]
@olegklimov is this bounty still available

**Comment by olegklimov** [comment]
De-prioritized for now, let's close it not to confuse people.
